rfc/previews-using-fetaure-flags.md

```diff
@@ -0,0 +1,306 @@
+    steps:
+      - uses: wundergraph/feature-flags@main
+        id: ff
+        with:
+          config: cosmo.yaml # Default to .github/cosmo.yaml
+          destroy: true # Only destroy the feature flag + subgraphs from cosmo.yaml
+      - run: |
+          echo "Feature flag and the feature subgraphs have been destroyed. You can now unprovision the subgraph service."
+---
+# config
+# .github/cosmo.yaml
+namespace: 'staging'
+feature_flags:
+  - name: 'my-feature-flag-{PR_NUMBER}'
+    labels:
+      - 'my-feature-flag-label'
+
+subgraphs:
+  - name: 'subgraph1'
+    schema: 'my-project/my-subgraph/subgraph1.graphql'
+    routing_url: 'http://mycompany-subgraph1-{PR_NUMBER}.com/graphql' # Supports templating, only PR_NUMBER is supported
+```
+
+### Requirements
+
+- Make sure the labels of the feature flag match that of the federated graph for which the preview is being created.
+- Make sure that the subgraphs being changed are a part of the federated graph.
+- Make sure that the changed subgraphs are a part of the config, if not present the featur subgraph for that subgraph will not be created.
+- All the subgraphs in the config have to be deployed and present on Cosmo.
+- The `cosmo.yaml` file should be present in the `.github` directory before the PR is labeled with `cosmo-preview`.
+- All the subgraphs changed in this PR should be deployed at the proposed routing_url in the CI for the preview to work.
+
+### Implementation
+
+The implementation is the same as Proposal-1, the only difference is that the feature flag is created when the PR is labeled with `cosmo-preview` and destroyed when the PR is unlabeled with `cosmo-preview` or when the pr is closed.
+
+## Pros
+
+- As the feature flag is created only when the PR is labeled with `cosmo-preview`, the feature flag is not created for every PR. This can be useful as all the PRs might not need a preview.
+
+### Limitations
+
+- The comso.yaml file should not be changed after the PR is labeled with `cosmo-preview`, if changed for the preview to work as expected, the PR has to be unlabeled and relabeled with `cosmo-preview`.
+
+
+## Proposal-3
+
+### Introduction
+
+This proposal simplifies the process by handling a single feature subgraph per feature flag and using direct parameters instead of a configuration file. This approach is more straightforward but less flexible, suitable for cases where only one subgraph needs to be previewed per PR.
+
+```yaml
+# feature-flags-preview.yaml
+on:
+  pull_request:
+    paths:
+      - "services/subgraph1/**"
+    types: [ opened, synchronize, closed, reopened ]
+
+jobs:
+  create:
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.action == 'opened' || github.event.pull_request.action == 'reopened'
+    steps:
+      # 1. Create & Publish feature subgraph based on cosmo.yaml
+      # 2. Create a new feature flag
+      - uses: wundergraph/feature-flags@main
+        id: ff
+        with:
+          namespace: "production"
```

Comment:
It doesn't really look simpler to me.
We can also provide a JSON Schema for `.github/comso.yaml` to make the configuration easier

Reply:
Do you mean configuring using the 3rd proposal is not simple?

Reply:
Regarding the JSON Schema for `.github/comso.yaml`, we can provide that

Reply:
> Do you mean configuring using the 3rd proposal is not simple?

I don't think it's simpler than using `.github/cosmo.yaml`, you still have to configure a lot of things, just in a different place. So it's maybe marginally simpler with the downsides of not being able to handle multiple subgraphs.

Reply:
yep